### PR TITLE
Adds support for SDWebImage and takes preference over JMImageCache

### PR DESCRIPTION
**Pull request by markrickert**
Initial PR. I'd like to add some tests for the new `:prepare_for_reuse` method call.

### CODE REVIEW

File: lib/ProMotion/table/cell/table_view_cell_module.rb [hunk 1]
```diff
@@ -89,12 +100,26 @@ def set_accessory_type
       self.accessoryType = map_accessory_type_symbol(data_cell[:accessory_type]) if data_cell[:accessory_type]
     end
 
+    def prepareForReuse
+      if @remote_image_operation && @remote_image_operation.respond_to?(:cancel)
+        @remote_image_operation.cancel
+        @remote_image_operation = nil
+      end
+      self.send(:prepare_for_reuse) if self.respond_to?(:prepare_for_reuse)
```
Comment (jamonholmgren) [inline lib/ProMotion/table/cell/table_view_cell_module.rb]: Need a call to `super` here.

> . If you override this method, you must be sure to invoke the superclass implementation.

Comment (markrickert) [inline lib/ProMotion/table/cell/table_view_cell_module.rb]: good call. I'm jsut finishing the tests to make sure that this method calls `prepare_for_reuse` on the cell and that `prepare_for_reuse` is called BEFORE `on_reuse`

File: lib/ProMotion/table/cell/table_view_cell_module.rb [hunk 2]
```diff
@@ -89,12 +100,26 @@ def set_accessory_type
...
+    end
+
   private
 
+    def sd_web_image?
+      return false if RUBYMOTION_ENV == 'test'
+      return true if defined?(SDWebImageManager) == 'constant' && SDWebImageManager.class == Class
+      false
+    end
```
Comment (jamonholmgren) [inline lib/ProMotion/table/cell/table_view_cell_module.rb]: Could probably simplify:

```ruby
def sd_web_image?
  return false if RUBYMOTION_ENV == 'test'
  !!defined?(SDWebImageManager)
end
```

**Comment by markrickert** [comment]
`302 specifications (610 requirements), 0 failures, 0 errors`

**Comment by markrickert** [comment]
Closes #683

File: lib/ProMotion/table/cell/table_view_cell_module.rb (hunk 1 above)
Comment (jamonholmgren) [inline lib/ProMotion/table/cell/table_view_cell_module.rb]: :+1:

**Comment by jamonholmgren** [comment]
Good work!

**Comment by markrickert** [comment]
Thanks man!

**Pull request merged**